src/common/transformations/src/transformations/common_optimizations/activations_scaling.cpp

```diff
@@ -0,0 +1,222 @@
+            scaled_down_nodes.insert(node->get_friendly_name());
+            scaled_down_nodes.insert(scale_down->get_friendly_name());
+            continue;
+        }
+
+        auto const_node = std::dynamic_pointer_cast<ov::op::v0::Constant>(node);
+        if (const_node) {
+            constant_nodes.insert(node->get_friendly_name());
+            continue;
+        }
+
+        auto group_norm_node = std::dynamic_pointer_cast<ov::op::v12::GroupNormalization>(node);
+        auto mvn_node = std::dynamic_pointer_cast<ov::op::v6::MVN>(node);
+        if (group_norm_node || mvn_node) {
+            normal_nodes.insert(node->get_friendly_name());
+            continue;
+        }
+
+        size_t num_scaled_down_inputs = 0;
+        size_t num_const_inputs = 0;
+        size_t num_normal_inputs = 0;
+        for (auto& dep: node->inputs()) {
+            auto dep_name = dep.get_source_output().get_node_shared_ptr()->get_friendly_name();
+
+            if (scaled_down_nodes.find(dep_name) != scaled_down_nodes.end()) {
+                num_scaled_down_inputs += 1;
+                continue;
+            }
+            if (constant_nodes.find(dep_name) != constant_nodes.end()) {
+                num_const_inputs += 1;
+                continue;
+            }
+            if (normal_nodes.find(dep_name) != normal_nodes.end()) {
+                num_normal_inputs += 1;
+                continue;
+            }
+        }
+
+        if (node->get_input_size() > 0) {
+            if (num_const_inputs == node->get_input_size()) {
+                constant_nodes.insert(node->get_friendly_name());
+                continue;
+            }
+            if ((num_const_inputs + num_normal_inputs) == node->get_input_size()) {
+                normal_nodes.insert(node->get_friendly_name());
+                continue;
+            }
+        }
+
+        if (num_scaled_down_inputs == 0) {
+            continue;
+        }
+
+        auto result = std::dynamic_pointer_cast<ov::op::v0::Result>(node);
+        if (result && num_scaled_down_inputs == 1) {
+            auto dep = node->input(0);
+            std::shared_ptr<ov::Node> scale_const = (dep.get_element_type() == ov::element::f16) ?
+                                                    scale_const_f16 : scale_const_f32;
+            auto scale_up = std::make_shared<ov::op::v1::Multiply>(dep.get_source_output(), scale_const->output(0));
+            dep.replace_source_output(scale_up->output(0));
+        }
+
+        //    input0         input1            input0         input1
+        // (scaled_down)  (non-scaled)      (scaled_down)  (non-scaled)
+        //          \         /                      \         /
+        //           \       /         ==>            \    scale_down
+        //            \     /                          \     /
+        //              add                              add
+        auto add = std::dynamic_pointer_cast<ov::op::v1::Add>(node);
```

Comment:
I think what this pass should do in the first place is following:
in_f16 -> conv/matmul/any_other_needed_op -> out_f16
in_f16 -> Multiply(down)> conv/matmul/any_other_needed_op -> Multiply(up) -> out_f16

Which works well with matcher pass concept. Then we need to optimize case
Multiply(up) -> Multiply(down) to eliminate those. I think it can be achieved using different approaches:
1. Run some common pass which merged chain of Multiply with scalar into single node (likely we have such opt already). + run NopElimination
2. Respect Multiply(up) in pattern and do different handling in the callback. For instance, you match optional(Multiply_up) -> Conv pattern. If Multiply is captured, then you just move this Multiply_up node after Convolution. If not, then you insert a pair Multiply down + up around conv.
3. Implement custom pass which eliminates Multiply down -> Multiply up sequences

IMO, option 2 may be a good solution

Reply:
Update: Suggestion is to reuse LPT for scale nodes optimization. I.e. at first we insert Multiplies around required ops, then run LPT subset which propagates Multiply (in case of int8 it's dequantization, but it's still Multiply op). Hopefully, we can reuse LPT passes w/o modifications.
Also, `MoveEltwiseUpThroughDataMov` pass will help to move those scales up in the graph.

Reply:
I updated `ActivationsScaling` as you reviewed.
First, it inserts `Multiply` nodes around Conv/MatMul, and eliminates these newly added `Multiply` nodes by a series of optimization passes. One of them is `LinOpSequenceFusion` from LPT.